sch_cake.c

```diff
@@ -1619,12 +1619,20 @@ static unsigned int cake_drop(struct Qdisc *sch, struct sk_buff **to_free)
 	return idx + (tin << 16);
 }
 
+static inline __be16 cake_skb_proto(const struct sk_buff *skb)
+{
+	if (skb->protocol == htons(ETH_P_8021Q))
+		return vlan_eth_hdr(skb)->h_vlan_encapsulated_proto;
```

Comment:
You can't dereference a packet header without a preceeding pskb_may_pull() check.

Reply:
Ok, will fix it.